src/ape/managers/chain.py

```diff
@@ -92,15 +97,72 @@ def __iter__(self) -> Iterator[BlockAPI]:
 
         return self.range()
 
+    def query(
+        self,
+        columns: Union[str, List[str]],
+        start_block: int = 0,
+        stop_block: Optional[int] = None,
+        engine_to_use: Optional[QueryAPI] = None,
+    ) -> pd.DataFrame:
+        """
+        A method for querying blocks and returning a pandas DataFrame. If you
+        do not provide a starting block, the 0 block is assumed. If you do not
+        provide a stopping block, the last block is assumed. You can pass engine_to_use
+        to short-circuit engine selection.
+
+        Raises:
+            :class:`~ape.exceptions.ChainError`: When ``stop_block`` is greater
+              than the chain length.
+
+        Args:
+            columns (Union[str, List[str]]): columns in the DataFrame to return
+            start_block (int): The first block, by number, to include in the
+              range. Defaults to 0.
+            stop_block (Optional[int]): The block number to stop before. Defaults
+              to the lastest block.
+            engine_to_use (Optional[QueryAPI]): query engine to use, bypasses query
+              engine selection algorithm.
+
+        Returns:
+            pandas.DataFrame
+        """
+
+        if stop_block is None:
+            stop_block = len(self)
```

Comment:
Isn't `len(self)` technically the block number of the block after the last one?

Reply:
@unparalleled-js thanks for catching this!